src/modules/helpers/eval.ts

```diff
@@ -133,9 +134,21 @@ function evalProcessFunction(
   return [
     index + (nextChar === '.' ? 2 : 1), // one for the closing bracket, one for the dot
     entrypoints.map((entrypoint): unknown =>
-      // TODO @ST-DDT 2023-12-11: Replace in v9
+      // TODO @ST-DDT 2023-12-11: Replace in v10
       // typeof entrypoint === 'function' ? entrypoint(...params) : undefined
-      typeof entrypoint === 'function' ? entrypoint(...params) : entrypoint
+      {
+        if (typeof entrypoint === 'function') {
+          return entrypoint(...params);
+        }
+
+        deprecated({
+          deprecated: 'Calling functions on non-functions',
```

Comment:
Maybe:

```suggestion
          deprecated: 'Invoking expressions which are not functions',
```

Should we log the expression which got falsely invoked? That would help clearify the warning for users. Since this block is in a "map" function, it might be triggered multiple times without additional value.

Reply:
I agree that the current error is probably going to be pretty confusing to end users

If I previously had code like 
```
faker.helpers.fake('{{person.first_name() {{person.lastName()}}'); I will probably need guidance as to what is wrong.

Reply:
Should be fixed now.